app/controllers/miq_ae_tools_controller.rb

```diff
@@ -194,44 +194,55 @@ def review_import
   def retrieve_git_datastore
     redirect_options = {:action => :review_git_import}
     git_url = params[:git_url]
-    verify_ssl = params[:git_verify_ssl] == "true" ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE 
+    verify_ssl = params[:git_verify_ssl] == "true" ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
+    new_git_repo = nil
 
     if git_url.blank?
       add_flash(_("Please provide a valid git URL"), :error)
     elsif !GitBasedDomainImportService.available?
       add_flash(_("Please enable the git owner role in order to import git repositories"), :error)
     else
-      if GitRepository.exists?(:url => git_url)
-        flash_message = _("This repository has been used previously for imports; If you use the same domain it will get deleted and recreated")
-        status = :warning
-      else
-        flash_message = _("Successfully found git repository, please choose a branch or tag")
-        status = :success
-      end
+      begin
+        git_repo = GitRepository.find_by(:url => git_url)
+        unless git_repo
+          git_repo = GitRepository.create!(:url => git_url)
+          new_git_repo = git_repo
```

Comment:
The `find_or_create_by!` method accepts a block, which is only passed down to create.  [See here](http://apidock.com/rails/v4.2.7/ActiveRecord/Relation/find_or_create_by)

If you treat `new_git_repo` as a boolean you can do this in one line:
```ruby
git_repo = GitRepository.find_or_create_by!(:url => git_url) { new_git_repo = true }
```

Then in the rescue block below change to `git_repo.destroy if new_git_repo`